module/users/helper.js

```diff
@@ -300,25 +351,33 @@ module.exports = class UserHelper {
                             solutionData => solutionData.externalId === solution.externalId
                         );
 
-                        if( solutionIndex < 0 ) {
-
-                            let solutionOrObservationInformation = 
-                            users[user].isObservation ? 
-                            _observationInformation(
-                                program,
-                                users[user],
-                                solution
-                            ) : _solutionInformation(program,solution); 
-                            
-                            solutionOrObservationInformation["entities"] = [];
-                            result[programIndex].solutions.push(
-                                solutionOrObservationInformation
-                            );
+                        if (solutionIndex < 0) {
+
+                            let solutionOrObservationInformation =
+                                    users[user].isObservation || users[user].isSurvey ?
+                                        _observationInformation(
+                                            program,
+                                            users[user],
+                                            solution
+                                        ) : _solutionInformation(program, solution);
+                                
+                                   solutionOrObservationInformation["entities"] = [];
+                                
+                                result[programIndex].solutions.push(
+                                    solutionOrObservationInformation
+                                );
 
                             solutionIndex = 
                             result[programIndex].solutions.length - 1;
                         }
-
+                        
+                        if (users[user].isSurvey) {
```

Comment:
@deepakr123 - Please remove this condition.

Reply:
resolved